# Summary of Scraped Data

In [1]:
import pandas as pd

In [2]:
pharm_df = pd.read_csv("scraped_pubchem_data_final.csv")

In [3]:
compound_df = pharm_df[pharm_df["Compound/Substance"] == "COMPOUND"]
compound_df.describe()

Original Molecule            Parsed Molecule  \
count                        2418                       2418   
unique                       2418                       2418   
top     1-2-PROPANEDIOL_DIACETATE  1,2 PROPANEDIOL DIACETATE   
freq                            1                          1   

                                                 Link Result Type      CAS  \
count                                            2418        2418     2303   
unique                                           2329           2     2208   
top     https://pubchem.ncbi.nlm.nih.gov/compound/702    FEATURED  64-17-5   
freq                                               10        2236       10   

                                           Deprecated CAS Compound/Substance  
count                                                1060               2418  
unique                                                978                  1  
top     121182-78-3, 8000-16-6, 8024-45-1, 8024-45-1, ...           COMPOUND  
freq                                                   10               2418

In [1]:
substance_df = pharm_df[pharm_df["Compound/Substance"] == "SUBSTANCE"]
substance_df.describe()

NameError: name 'pharm_df' is not defined

In [5]:
missing_df = pharm_df[pharm_df["Compound/Substance"].isnull()]
missing_df.describe()

Original Molecule  \
count                                    164   
unique                                   164   
top     2-3-(2-IODOPROPYLIDENEDIOXY)PROPANOL   
freq                                       1   

                             Parsed Molecule  \
count                                    164   
unique                                   164   
top     2,3-(2 IODOPROPYLIDENEDIOXY)PROPANOL   
freq                                       1   

                                            Link Result Type  CAS  \
count                                        115         164    0   
unique                                       114           3    0   
top     https://pubmed.ncbi.nlm.nih.gov/10292123    RELEVANT  NaN   
freq                                           2         106  NaN   

       Deprecated CAS Compound/Substance  
count               0                  0  
unique              0                  0  
top               NaN                NaN  
freq              NaN                NaN

In [6]:
missing_cas_df = pharm_df[pharm_df["CAS"].isnull()]
missing_cas_df.describe()

Original Molecule  \
count                                    584   
unique                                   584   
top     2-3-(2-IODOPROPYLIDENEDIOXY)PROPANOL   
freq                                       1   

                             Parsed Molecule  \
count                                    584   
unique                                   584   
top     2,3-(2 IODOPROPYLIDENEDIOXY)PROPANOL   
freq                                       1   

                                                     Link Result Type  CAS  \
count                                                 535         584    0   
unique                                                519           3    0   
top     https://pubchem.ncbi.nlm.nih.gov/substance/483...    RELEVANT  NaN   
freq                                                    5         425  NaN   

       Deprecated CAS Compound/Substance  
count               0                420  
unique              0                  2  
top               NaN          SUBSTANCE  
freq              NaN                305

## Summary:
- There are a total of 584 molecules with missing CAS values.
    - Of the 584, 164 are unclassified (Compound v.s. Substance).
    - Another 305 are substances.
    - The remaining 115 are compounds without a CAS.